editor/editor_node.cpp

```diff
@@ -1089,14 +1093,21 @@ void EditorNode::_save_scene(String p_file, int idx) {
 
 void EditorNode::_save_all_scenes() {
 
+	int current = editor_data.get_edited_scene();
 	for (int i = 0; i < editor_data.get_edited_scene_count(); i++) {
 		Node *scene = editor_data.get_edited_scene_root(i);
-		if (scene && scene->get_filename() != "") {
-			if (i != editor_data.get_edited_scene())
-				_save_scene(scene->get_filename(), i);
-			else
-				_save_scene_with_preview(scene->get_filename());
-		} // else: ignore new scenes
+
+		if (scene) {
+			bool unsaved = (i == current) ? saved_version != editor_data.get_undo_redo().get_version() : editor_data.get_scene_version(i) != 0;
```

Comment:
I think this logic is already implemented in `_next_unsaved_scene`.

Reply:
I understand you want to use `_next_unsaved_scene` here to check if it must be saved or not instead repeating the `unsave` code. I will update the commit with that.